pkg/controller/reconciler/nameserverdelegator_reconciler.go

```diff
@@ -36,31 +38,46 @@ func (z *nameserverDelegationReconciler) SetStateHandlers(handlers *clientCore.S
 
 // Reconcile knows how to do what is necessary to ensure the desired state is achieved
 func (z *nameserverDelegationReconciler) Reconcile(node *resourcetree.ResourceNode) (result ReconcilationResult, err error) {
-	switch node.State {
-	case resourcetree.ResourceNodeStatePresent:
-		hz, err := z.stateHandlers.Domain.GetPrimaryHostedZone()
-		if err != nil {
-			return result, fmt.Errorf("getting primary hosted zone: %w", err)
+	hz, err := z.stateHandlers.Domain.GetPrimaryHostedZone()
+	if err != nil {
+		return result, fmt.Errorf("getting primary hosted zone: %w", err)
+	}
+
+	var labels []string
+
+	if z.commonMetadata.Declaration.Experimental != nil {
```

Comment:
If we make the Experimental struct a non pointer, we wont have to check for nil. Is there any advantage of using a pointer? (except for microscopic memory usage reduction)

Reply:
Since we are using `omitempty`, we need to use a pointer, otherwise it will create an empty struct